# the following function will give you a NameError

def name_error():
    a = 2
    
print (a)

# the following function will give you a TypeError

def type_error():
    b = '4' + 3

type_error()

# the following function will give you a SyntaxError

def syntax_error()
   c = 4

syntax_error()

# the following function will give you a AttributeError

def attribute_error():
    d = 3

print(attribute_error.d)
